jcli/src/jcli_lib/transaction/simplified.rs

```diff
@@ -0,0 +1,161 @@
+
+    #[structopt(default_value = "block-0.bin")]
+    pub block0_path: PathBuf,
+
+    #[structopt(flatten)]
+    rest_args: RestArgs,
+}
+
+impl SimplifiedTransaction {
+    pub fn exec(self) -> Result<(), Error> {
+        let secret_key = read_ed25519_secret_key_from_file(&self.secret)?;
+        let (receiver_secret_key, receiver_address) = create_receiver_secret_key_and_address()?;
+        let fragment_id = simplified_transaction(
+            self.faucet_address,
+            receiver_address,
+            secret_key,
+            self.value,
+            self.fee,
+            self.block0_path,
+            self.rest_args.clone(),
+            self.change,
+        )?;
+        println!("{}", fragment_id);
+        println!(
+            "Private key of receiver (to revert transaction for testing purposes): {}",
+            receiver_secret_key.to_bech32_str()
+        );
+        println!("To see if transaction is in block use:");
+        println!("jcli rest v0 message logs -h {}", &self.rest_args.host);
+        println!("To check new account balance :");
+        println!(
+            "jcli  rest v0 account get  $RECEIVER_ADDR -h {}",
+            &self.rest_args.host
+        );
+        Ok(())
+    }
+}
+
+fn create_new_private_key() -> Result<SecretKey<Ed25519Extended>, Error> {
+    let rng = ChaChaRng::from_rng(OsRng)?;
+    let key = SecretKey::<Ed25519Extended>::generate(rng);
+    Ok(key)
+}
+
+fn create_receiver_address(sk: &SecretKey<Ed25519Extended>) -> interfaces::Address {
+    let pk = sk.to_public();
+    make_address(pk)
+}
+
+fn make_address(pk: PublicKey<Ed25519>) -> interfaces::Address {
+    chain_addr::Address(chain_addr::Discrimination::Test, Kind::Account(pk)).into()
+}
+
+fn create_receiver_secret_key_and_address(
+) -> Result<(SecretKey<Ed25519Extended>, interfaces::Address), Error> {
+    let sk = create_new_private_key()?;
+    let address = create_receiver_address(&sk);
+    Ok((sk, address))
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn simplified_transaction(
+    faucet_address: interfaces::Address,
+    receiver_address: interfaces::Address,
+    secret_key: EitherEd25519SecretKey,
+    value: interfaces::Value,
+    fee: common::CommonFees,
+    block0_file: PathBuf,
+    rest_args: RestArgs,
+    change: Option<interfaces::Address>,
+) -> Result<String, Error> {
+    let mut transaction = Staging::new();
+
+    // add account
+    transaction::add_account::add_account(faucet_address.clone(), value, &mut transaction)?;
+
+    // add output
+    transaction.add_output(Output {
+        address: receiver_address.into(),
+        value: value.into(),
+    })?;
+
+    //finalize
+    transaction::finalize::finalize(fee, change, &mut transaction)?;
+
+    // get transaction and block0 ids
+    let transaction_sign_data_hash = transaction.transaction_sign_data_hash();
+    let block0 = load_block(open_block_file(&Some(block0_file))?)?;
+    let block0_hash = block0.header.id();
```

Comment:
If the only point of using `block0` is to take its hash, then just use its hash as a command argument.

Reply:
Cool. Just tried to stick to the original script here (and in other places).